src/plugins/discover/public/__mocks__/index_patterns.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPatternsService } from '../../../data/common';
+import { indexPatternMock } from './index_pattern';
+
+export const indexPatternsMock = ({
+  getCache: () => {
+    return [indexPatternMock];
+  },
+  get: (id: string) => {
+    if (id === 'the-index-pattern-id') {
+      return indexPatternMock;
+    }
```

Comment:
for the case when the id is different? of course we can build up on this very basic mock, will be useful in other tests, however the original code here provides different error handling, so mocking this would work differently